### ISSUE TITLE
Documentation transcribe callbacks

### ISSUE THREAD

**Issue by yzabalotski** [aws/aws-sdk-cpp#1234]
https://github.com/aws/aws-sdk-cpp/blob/4dfce9c92c7f9f95fcd13c8973df7bfc059e419d/aws-cpp-sdk-transcribestreaming-integration-tests/TranscribeTests.cpp#L137

Hi, it will be nice to have a couple of words about how does transcribe streaming work under the hood ... at least for me :)

We have TranscriptEventCallback, OnStreamReady, OnResponseCallback.
Can be two instance of TranscriptEventCallback executed simultaneously? or first call of TranscriptEventCallback should be finished before second one?

As I know, transcribe client uses http2, the client at separate thread o
pens a tcp connection and sends init request to transcribe service to be able sign further audio requests. After the client got init response, OnStreamReady is called (here we put our audio events), at the end of OnStreamReady event with empty body should be generated (for correct closing ?). Separately, at some thread, TranscriptEventCallback is called when service responds with async events. And finally OnResponseCallback is called. Am I wrong? 

Please, add some description about thread safety and execution order of these thee callbacks at least at integration test.
